### PR TITLE
Generate operation classes

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: test/TestProjects/TypeSchemaMapping/Generated/MainClient.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 namespace CustomNamespace
 {
-    /// <summary> The Main service client. </summary>
+    /// <summary> The Service service client. </summary>
```
Comment (MiYanni) [inline test/TestProjects/TypeSchemaMapping/Generated/MainClient.cs:18]: Looks like something goofed up here.

File: src/assets/Generator.Shared/ArmOperationHelpers.cs [hunk 2]
```diff
@@ -13,344 +13,344 @@
...
-    /// Helper methods for ARM long-running operations.
+    /// This implements the ARM scenarios for LROs. It is highly recommended to read the ARM spec prior to modifying this code:
+    /// https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations
+    /// Other reference documents include:
+    /// https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
+    /// https://github.com/Azure/adx-documentation-pr/blob/master/sdks/LRO/LRO_AzureSDK.md
     /// </summary>
-    internal static class ArmOperationHelpers
+    /// <typeparam name="T">The final result of the LRO.</typeparam>
+    internal class ArmOperationHelpers<T>
```
Comment (MiYanni) [inline src/assets/Generator.Shared/ArmOperationHelpers.cs:23]: I'm a little confused by this. It is essentially what was `ArmOperation` but not a type of `Operation<T>`. Is there a reason for that? You're already using this as a composition-based operation field inside an operation.

File: test/TestServerProjects/paging/Generated/PagingGetMultiplePagesLROOperation.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using paging.Models;
+
+namespace paging
+{
+    /// <summary> A long-running paging operation that includes a nextLink that has 10 pages. </summary>
+    public partial class PagingGetMultiplePagesLROOperation : Operation<ProductResult>, IOperationSource<ProductResult>
```
Comment (MiYanni) [inline test/TestServerProjects/paging/Generated/PagingGetMultiplePagesLROOperation.cs:20]: `LROOperation` 🤣 Good ol' ATM Machine. 😋

File: src/assets/Generator.Shared/ArmOperationHelpers.cs (hunk 2 above)
Comment (pakrym) [reply src/assets/Generator.Shared/ArmOperationHelpers.cs:23]: There is just no reason for it to implement Operation<T> and I don't want anyone to accidentally use it directly.

File: test/TestProjects/TypeSchemaMapping/Generated/MainClient.cs (hunk 1 above)
Comment (pakrym) [reply test/TestProjects/TypeSchemaMapping/Generated/MainClient.cs]: Fixes.

File: src/AutoRest.CSharp.V3/Generation/Writers/OperationWriter.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+using AutoRest.CSharp.V3.Output.Models.Requests;
+using AutoRest.CSharp.V3.Output.Models.Serialization;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Request = Azure.Core.Request;
+
+namespace AutoRest.CSharp.V3.Generation.Writers
+{
+    internal class OperationWriter
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/AutoRest.CSharp.V3/Generation/Writers/OperationWriter.cs]: Renamed to LongRunningOperationWriter.

**Pull request merged**